src/Command/Pull/PullCommandBase.php

```diff
@@ -861,7 +864,7 @@ protected function determineEnvironment(InputInterface $input, OutputInterface $
    */
   protected function checkEnvironmentPhpVersions(EnvironmentResponse $environment): void {
     if (!$this->environmentPhpVersionMatches($environment)) {
-      $this->io->warning("You are using PHP version {$this->getCurrentPhpVersion()} but the upstream environment {$environment->label} is using PHP version {$environment->configuration->php->version}");
+      $this->io->warning("You are using PHP version {$this->getIdePhpVersion()} but the upstream environment {$environment->label} is using PHP version {$environment->configuration->php->version}");
```

Comment:
Here's the meat of this PR, calling `getIdePhpVersion` which reads the PHP version from a config file. This should be safe because pull:code can only be called from within an IDE, and this file should always exist.